distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Extensions.enso

```diff
@@ -229,11 +229,41 @@ Text.characters self =
          example_match =
              regex = "a[ab]c"
              "aabbbbccccaabcaaaa".match regex == "abc"
-Text.match : Text -> (Matching_Mode.First | Matching_Mode.Last) -> (Text_Matcher | Regex_Matcher) -> Text | Nothing
-Text.match self term mode=Matching_Mode.First matcher=Regex_Matcher.Value =
-    case self.locate term mode matcher of
-        Nothing -> Nothing
-        span -> span.text
+
+   ! Last Match in Regex Mode
+     Regex always performs the search from the front and matching the last
+     occurrence means selecting the last of the matches while still generating
+     matches from the beginning. This will lead to slightly different behavior
+     for overlapping occurrences of a pattern in Regex mode than in exact text
+     matching mode where the matches are searched for from the back.
+
+   > Example
+     Comparing Matching in Last Mode in Regex and Text mode
+
+         "aaa".match "aa" mode=Matching_Mode.Last matcher=Text_Matcher == "aaa"
+         "aaa".match "aa" mode=Matching_Mode.Last matcher=Regex_Matcher == "aaa"
+
+         "aaa aaa".match "aa" mode=Matching_Mode.Last matcher=Text_Matcher == "aaa aaa"
+         "aaa aaa".match "aa" mode=Matching_Mode.Last matcher=Regex_Matcher == "aaa aaa"
```

Comment:
Once we stripped off the locations, this examples does not really show any difference - the return values are exactly the same. I think if we still want to illustrate the difference we'd need to do:
```suggestion
   > Example
     Comparing Matching in Last Mode in Regex and Text mode

         "aAa".match "aa" mode=Matching_Mode.Last matcher=Text_Matcher.Case_Insensitive == "Aa"
         "aAa".match "aa" mode=Matching_Mode.Last matcher=(Regex_Matcher.Value case_sensitivity=Case_Sensitivity.Insensitive) == "aA"
```

btw. the example seem to have been out of date, so they need updating anyway. (They were already out of date before your changes, seems that we forgot to update them when reworking the matchers)

(I did not run the examples above, please check if they return what is expected, in fact - please just add them to the tests, replacing `==` with ` . should_equal`.